setup/classes/class.ilSetupGUI.php

```diff
@@ -4329,19 +4330,24 @@ function cloneSaveSource()
 
 		if ($this->form->checkInput())
 		{
+			$error = '';
 			if ($this->form->getInput("iamsure") != "1")
 			{
-				$message = $this->lng->txt("clone_youmustcheckiamsure");
-				$this->setup->raiseError($message,$this->setup->error_obj->MESSAGE);
+				$error = $this->lng->txt('clone_youmustcheckiamsure');
 			}
 			if (!$this->setup->cloneFromSource($this->form->getInput("source")))
 			{
-				$message = $this->lng->txt("clone_error");
-				$this->setup->raiseError($message . " -> " . $this->setup->error,$this->setup->error_obj->MESSAGE);
+				$error = $this->lng->txt('clone_error') . ' -> ' . $this->setup->error;
 			}
 
 			ilUtil::sendInfo($this->lng->txt("client_cloned"),true);
 			// ilUtil::redirect("setup.php");
+
+			if (0 === strlen($error)) {
+				\ilUtil::sendInfo($this->lng->txt('client_cloned'));
+			} else {
+				\ilUtil::sendFailure($error);
```

Comment:
This does not look correct. `sendInfo` might be called twice. `sendFailure` could be called directly after the setting of `$error` like in the other places. `redirect` seems to be missing after the `sendFailure`.